### ISSUE TITLE
DB::OpenAsSecondary hangs if primary DB not available

### ISSUE THREAD

**Issue by rohitjoshi** [facebook/rocksdb#7031]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://groups.google.com/forum/#!forum/rocksdb or https://www.facebook.com/groups/rocksdb.dev

### Expected behavior
While opening DB using [DB::OpenAsSecondary](https://github.com/facebook/rocksdb/wiki/Secondary-instance) and if primary DB path not available, it should return an error.

Similarly Read() operation is blocked when key not found in the secondary db log files and primary not available after successful startup.

Primary DB may not available due to NFS mount not available temporarily  / network issues.

### Actual behavior
DB::OpenAsSecondary() function call hangs and don't not return any error.

### Steps to reproduce the behavior

```
const std::string kDbPath = "/tmp/notexist";
Options options;
options.max_open_files = -1;
// Secondary instance needs its own directory to store info logs (LOG)
const std::string kSecondaryPath = "/tmp/rocksdb_secondary/";
DB* db_secondary = nullptr;
Status s = DB::OpenAsSecondary(options, kDbPath, kSecondaryPath, &db_secondary);
```

**Comment by mrambacher** [comment]
I tracked this down a little but am not sure what the correct fix is.  The problem is that MaybeSwitchManifest loops forever on "PathNotFound".  So if the path to the primary DB is non-existent, this code will loop forever waiting for it to come into existence.

**Comment by riversand963** [comment]
We need more sophisticated error handling here. Depending on whether the underlying error is retryable, we can either report failure or retry.

**Comment by riversand963** [comment]
The original motivation for the loop on "PathNotFound" of the MANIFEST is to handle the case in which the primary repeatedly deletes old MANIFEST and switches to new one (an extreme and rare case).
For the case mentioned in this issue, we should not enter the loop. Maybe we can add a check for the existence of the MANIFEST before trying to open it. Even if the db directory becomes inaccessible in between, the next check for existence can catch it.
@mrambacher would you like to submit a fix?

**Comment by riversand963** [comment]
Fixed by #8200
